x-pack/plugins/reporting/server/browsers/install.ts

```diff
@@ -4,69 +4,56 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import fs from 'fs';
+import os from 'os';
 import path from 'path';
+import del from 'del';
+
 import * as Rx from 'rxjs';
-import { first } from 'rxjs/operators';
-import { promisify } from 'util';
-import { ReportingConfig } from '../';
 import { LevelLogger } from '../lib';
-import { BrowserDownload } from './';
 import { ensureBrowserDownloaded } from './download';
 // @ts-ignore
 import { md5 } from './download/checksum';
 // @ts-ignore
 import { extract } from './extract';
-
-const chmod = promisify(fs.chmod);
+import { paths } from './chromium/paths';
 
 interface Package {
   platforms: string[];
+  architecture: string;
 }
 
 /**
  * "install" a browser by type into installs path by extracting the downloaded
  * archive. If there is an error extracting the archive an `ExtractError` is thrown
  */
 export function installBrowser(
-  browser: BrowserDownload,
-  config: ReportingConfig,
-  logger: LevelLogger
+  logger: LevelLogger,
+  chromiumPath: string = path.resolve(__dirname, '../../chromium'),
+  platform: string = process.platform,
+  architecture: string = os.arch()
 ): { binaryPath$: Rx.Subject<string> } {
   const binaryPath$ = new Rx.Subject<string>();
   const backgroundInstall = async () => {
-    const captureConfig = config.get('capture');
-    const { autoDownload, type: browserType } = captureConfig.browser;
-    if (autoDownload) {
-      await ensureBrowserDownloaded(browserType, logger);
-    }
+    const pkg = paths.packages.find((p: Package) => {
+      return p.platforms.includes(platform) && p.architecture === architecture;
+    });
 
-    const pkg = browser.paths.packages.find((p: Package) => p.platforms.includes(process.platform));
     if (!pkg) {
-      throw new Error(`Unsupported platform: ${JSON.stringify(browser, null, 2)}`);
+      // TODO: validate this
+      throw new Error(`Unsupported platform: ${platform}-${architecture}`);
     }
 
-    const dataDir = await config.kbnConfig.get('path', 'data').pipe(first()).toPromise();
-    const binaryPath = path.join(dataDir, pkg.binaryRelativePath);
+    const binaryPath = path.join(chromiumPath, pkg.binaryRelativePath);
+    const binaryChecksum = await md5(binaryPath).catch(() => '');
 
-    try {
-      const binaryChecksum = await md5(binaryPath).catch(() => '');
+    if (binaryChecksum !== pkg.binaryChecksum) {
+      await ensureBrowserDownloaded(logger);
 
-      if (binaryChecksum !== pkg.binaryChecksum) {
-        const archive = path.join(browser.paths.archivesPath, pkg.archiveFilename);
-        logger.info(`Extracting [${archive}] to [${binaryPath}]`);
-        await extract(archive, dataDir);
-        await chmod(binaryPath, '755');
```

Comment:
is the chmod not needed anymore?

Reply:
Nope, we're no longer writing the Chromium debug log.